VERSIONING.md

```diff
@@ -0,0 +1,145 @@
+# Versioning details
+
+This follows the [OpenTelemetry versioning proposal](https://github.com/open-telemetry/oteps/pull/143/files)
+
+OpenTelemetry .NET follows [SemVer V2](https://semver.org/spec/v2.0.0.html)
+guidelines. This means that, for any stable packages released from this repo,
+all public APIs will remain backward compatible, unless a major version bump
```

Comment:
Consider adding a link to the [.NET breaking change rules](https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/breaking-change-rules.md#breaking-change-rules).